src/frontend/mame/info.cpp

```diff
@@ -352,41 +346,42 @@ void info_xml_creator::output(FILE *out, const std::function<bool(const char *sh
 //  the root element
 //-------------------------------------------------
 
-void info_xml_creator::output_header()
+void info_xml_creator::output_header(std::ostream &out)
 {
 	if (m_dtd)
 	{
 		// output the DTD
-		fprintf(m_output, "<?xml version=\"1.0\"?>\n");
+		out << "<?xml version=\"1.0\"?>" << std::endl;
 		std::string dtd(s_dtd_string);
 		strreplace(dtd, "__XML_ROOT__", XML_ROOT);
 		strreplace(dtd, "__XML_TOP__", XML_TOP);
 
-		fprintf(m_output, "%s\n\n", dtd.c_str());
+		out << dtd << std::endl << std::endl;
 	}
 
 	// top-level tag
-	fprintf(m_output, "<%s build=\"%s\" debug=\""
+	out << util::string_format("<%s build=\"%s\" debug=\""
 #ifdef MAME_DEBUG
 			"yes"
 #else
 			"no"
 #endif
-			"\" mameconfig=\"%d\">\n",
+			"\" mameconfig=\"%d\">",
 			XML_ROOT,
 			util::xml::normalize_string(emulator_info::get_build_version()),
 			CONFIG_VERSION);
+	out << std::endl;
```

Comment:
You shoudln't use `std::endl` unnecessarily.  It causes a flush as well as outputting a line ending.  You don't want to be flushing the buffers more than necessary when writing out a lot of content like this.